spark/src/main/scala/geotrellis/spark/io/avro/codecs/TemporalProjectedExtentCodec.scala

```diff
@@ -33,20 +33,34 @@ trait TemporalProjectedExtentCodec {
       .record("TemporalProjectedExtent").namespace("geotrellis.spark")
       .fields()
       .name("extent").`type`(extentCodec.schema).noDefault()
-      .name("epsg").`type`().intType().noDefault()
+      .name("epsg").`type`().optional().intType()
+      .name("proj4").`type`().optional().stringType()
       .name("instant").`type`().longType().noDefault()
       .endRecord()
 
     def encode(temporalProjectedExtent: TemporalProjectedExtent, rec: GenericRecord): Unit = {
+      val crs = temporalProjectedExtent.crs
       rec.put("extent", extentCodec.encode(temporalProjectedExtent.extent))
-      rec.put("epsg", temporalProjectedExtent.crs.epsgCode.get)
+      if(crs.epsgCode.isDefined) {
+        rec.put("epsg", crs.epsgCode.get)
+      }
+      else {
+        rec.put("proj4", crs.toProj4String)
+      }
```

Comment:
With Avro schema migration capabilities (which allow for aliasing), this might not have to be a breaking change. See: https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/io/avro/codecs/KeyCodecs.scala#L52